### PR TITLE
fix #17835(turn errors into warnings)

### LINKED ISSUE #17835: Docgen issue with runnableExamples on specific usage of anyIt template.
### Example
```nim
import std/sequtils

proc foo*(n: seq[int]): bool =
  result = anyIt(n, it == 0)
```

### Current Output
```
$ nim doc test.nim
Hint: used config file '/etc/nim/nim.cfg' [Conf]
Hint: used config file '/etc/nim/config.nims' [Conf]
Hint: used config file '/etc/nim/nimdoc.cfg' [Conf]
......
sequtils.nim(721, 3) Error: runnableExamples must appear before the first non-comment statement
```

### Additional Information
Any statement before `result =` will make the bug disappear.

This is causing disruptek/cps docs to not build.

```
$ nim -v
Nim Compiler Version 1.5.1 [Linux: amd64]
Compiled at 2021-04-24
Copyright (c) 2006-2021 by Andreas Rumpf

git hash: b667e288e27a09ce4f65166fc66a6b713350a9b6
active boot switches: -d:release -d:nimUseLinenoise -d:nativeStackTrace
```

Bisect shows that a807233aebcd3759bc3e21b450ed89e1eb6ddace is the first bad commit.

/cc @timotheecour @xflywind

### PR DESCRIPTION
**Pull request by ringabout**
fix #17835

> turn the error into a warning (temporarily), 

workaround the issue until the root cause  #16993 is fixed.

### CODE REVIEW

File: tests/nimdoc/t17835.nim [hunk 1]
```diff
@@ -0,0 +1,15 @@
+discard """
+  cmd: "nim doc $file"
+  action: "compile"
+  nimout: "t17835.nim(10, 3) Warning: runnableExamples must appear before the first non-comment statement [User]"
+  joinable: false
+"""
+
```
Comment (timotheecour) [inline tests/nimdoc/t17835.nim:7]: this test should fail once https://github.com/nim-lang/Nim/issues/16993 is fixed, but ok

**Review by timotheecour — DISMISSED** [review]

File: tests/nimdoc/t17835.nim (hunk 1 above)
Comment (ringabout) [reply tests/nimdoc/t17835.nim:7]: yeah, it will fail I think when #16993 is fixed, because then warnings won't show up.

**Comment by Araq** [comment]
The compiler shouldn't use "warnUser" itself. There should be a dedicated warning message.

**Comment by timotheecour** [comment]
true (though it's a temporary hotfix until https://github.com/nim-lang/Nim/issues/17835 is fixed, at which point the warning would be turned back to an error)

**Comment by timotheecour** [comment]
we should merge https://github.com/nim-lang/Nim/pull/18082 instead

**Pull request closed without merging**